sycl/include/sycl/ext/oneapi/matrix/matrix-intel.hpp

```diff
@@ -481,6 +481,8 @@ joint_matrix_store(Group sg,
                        Group, Tp, Use, NumRows, NumCols, Layout> &src,
                    multi_ptr<T, Space, IsDecorated> dst, size_t stride) {
 #if defined(__SYCL_DEVICE_ONLY__)
+  static_assert(Space != access::address_space::private_space,
+                "Joint Matrix doesn't support store to private memory!");
```

Comment:
Since joint matrix is not supported on host device, I thought it doesn't make sense to check and provide more specific diagnostic if the entire feature is not supported.